engine/execution/ingestion/loader/unexecuted_loader.go

```diff
@@ -231,8 +231,8 @@ func (e *UnexecutedLoader) pendingUnexecutedBlocks(ctx context.Context, finalize
 // if the EN is dynamically bootstrapped, the finalized blocks at height range:
 // [ sealedRoot.Height, finalizedRoot.Height - 1] can not be retrieved from
 // protocol state, but only from headers
-func (e *UnexecutedLoader) getHeaderByHeight(height uint64) (*flow.Header, error) {
+func (e *UnexecutedLoader) getHeaderByHeight(height uint64) (flow.Identifier, error) {
```

Comment:
```suggestion
func (e *UnexecutedLoader) getBlockIDByHeight(height uint64) (flow.Identifier, error) {
```

Reply:
same question here about not using `e.headers.BlockIDByHeight(height)` directly